### DISCUSSION TITLE
k8s watch resource

### DISCUSSION THREAD

**Discussion by lanzhiwang** [strimzi/strimzi-kafka-operator#5883] [General]
When I start strimzi operators, it starts to watch kafka cr. When I create kafka cr, Watcher will get the corresponding cr, and then execute related logic, such as creating a StatefulSet.

What I want to ask is that when I directly modify the spec of the StatefulSet, for example, modify the number of replicas from 3 to 4, eventually operators will still watch this change and restore the number of replicas of the StatefulSet to 3.

I don't think the operators directly watch the StatefulSet. Why can the operators perceive this change and finally restore the number of replicas of the StatefulSet to 3?

Is it because the ownerReferences is set in the StatefulSet, will the changes in the StatefulSet cause the operators to watch Kafka cr again?

**Comment by scholzj** [comment]
The Kubernetes watch is not the only trigger for the reconciliation events. It is also triggered periodically to check on things and on your brokers. So it will find out about it in one of the periodical reconciliations and fix the statefulset.
